api/serverless/getRepolinterResult/app.js

```diff
@@ -0,0 +1,67 @@
+const path = require('path');
+const repolinter = require('repolinter');
+const uuidv4 = require('uuid-v4');
+const rimraf = require('rimraf');
+const fs = require('fs');
+const git = require('simple-git')();
+
+async function runRepolinter(gitLink, ruleSet) {
+  const tmpDir = await path.resolve('/tmp', uuidv4());
+  await asyncGitClone(gitLink, tmpDir).catch(error => console.error(error));
+
+  const output = await asyncRunner(tmpDir, ruleSet);
+  rimraf(tmpDir, function() {});
+  return output;
+}
+
+async function asyncRunner(tmpDir, ruleSet) {
+  return new Promise(function(resolve, reject) {
+    const output = repolinter.lint(tmpDir, [], ruleSet);
+    if (output != null) {
+      resolve(output);
+    } else {
+      reject('error');
+    }
+  });
+}
+
+async function asyncGitClone(gitLink, tmpDir) {
+  return new Promise(function(resolve, reject) {
+    git.clone(gitLink, tmpDir, error => {
+      if (!error) {
+        resolve();
+      } else {
+        reject(error);
+      }
+    });
+  });
+}
+
+exports.lintRepo = async (event, context, callback) => {
+  process.env.PATH =
+    process.env.LAMBDA_TASK_ROOT + '/tmp/bin/usr/bin:' + process.env.PATH;
+  process.env.GIT_EXEC_PATH =
+    process.env.LAMBDA_TASK_ROOT + '/tmp/bin/usr/libexec/git-core';
+  const runningConfig = JSON.parse(event.body);
+  try {
+    const ret = await runRepolinter(
+      decodeURI(runningConfig.git_link.replace(/([%])/g, '/')),
```

Comment:
Replace was only there because earlier I was passing url as a path parameter vis GET call and had to replace "/" with something url friendly. I will remove it as this is now being taken care by as POST request.